code/script.js

```diff
@@ -229,97 +232,162 @@ const setSecret = () => {
     }
   } else if (category === 'other') {
-    // Similar to the one above
-  } else {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
-    } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      alert(
+        `Yes, the person is a ${value}! Keep all the smokers!`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => person [category].includes(value))
+    }
+    else {
+      alert(
+        `No, the person is not a ${value}! Remove all the smokers!`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
     }
   }
+  else if (category === 'hair'){
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
+    if (keep) {
+       alert( 
+         `Yes, the person has ${value} hair! Keep all people with ${value} hair!`
+       )
+       charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
+    } 
+    else {
+       alert( 
+        `No, the person doesnt have ${value} hair! Remove all people with ${value} hair`
+       )
+       charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
+    }
+  }
 
-  // Invoke a function to redraw the board with the remaining people.
+  else if (category === 'eyes'){
+    if (keep) {
+      alert(
+        `Yes, the person has ${value} eyes! Keep all the people with ${value} eyes!`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => person[category] == value)
+    }
+    else {
+      alert( 
+       `No, the person doesnt have ${value} eyes! Remove all people with ${value} eyes!`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
+  }
+  }
+  generateBoard()
 }
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (personToConfirm) => {
   // store the interaction from the player in a variable.
+  const confirming = confirm(`Do you really want to guess ${personToConfirm}?`)
+  if (confirming === true) {
+checkMyGuess(personToConfirm)
+  }
+  
+  else {
+    alert (`You Wuss!`)
+  }
   // remember the confirm() ?
   // If the player wants to guess, invoke the checkMyGuess function.
 }
 
 // If you confirm, this function is invoked
-const checkMyGuess = (personToCheck) => {
```

Comment:
Nice work spotting this error and using personToConfirm instead, I was really confused about this part.